### PR TITLE
[3.7.x] Article Manager Responsive Adjustments

### PR DESCRIPTION
**Pull request by ghost**
Pull Request for Issue #12789

### Summary of Changes
CSS Adjustments for screen sizes between mobile and large desktop sizes to address the larger button dimensions of the new interface design.

- Over 1200px there are no changes
- Under 1200px the padding and size of the button and button icon are slightly reduced
- Under 980px and 767px buttons which control functions available in the article list controls are hidden as the size decreases

There are a few minor break points where the buttons push on to two lines, this could be addressed through adding custom breakpoints, this PR uses only standard BS2 breakpoints.

We could also consider switching to the mobile toolbar at a higher screen size.

### Testing Instructions
View the article manager screen above 1200, below 1200, below 980, and below 767

### Expected result
article-manager-responsive

These changes are in a new file and are only applied to the Article Manager list view. 

### Documentation Changes Required
None

### CODE REVIEW

**Comment by dgt41** [comment]
I'll say no to this one as you are blindly removing buttons [and functionality that is unavailable without these buttons] on smaller screens, the scrolling toolbar that I proposed is far better approach IMHO

**Comment by brianteeman** [comment]
What criteria was applied to determine which functionality should be removed<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14501">issues.joomla.org/tracker/joomla-cms/14501</a>.</sub>

**Comment by brianteeman** [comment]
Also I might be wrong but this all seems to assume that the length of a button is the same in every language - which obviously it isn't. As you can see from this screenshot the toolbar buttons are much bigger in german

screen shot 2017-03-11 at 22 10 52<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14501">issues.joomla.org/tracker/joomla-cms/14501</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on b4373150961bb05f76f3da6b6642f86047903c12<br /><br />Applied PR - confirmed the new css is being loaded observed no visible changes. Expected to see toolbar buttons disappear as shown in the PR

not a css expert but you are targeting a class .com_content.view-articles that does not exist on the page you are trying to change the output of.
unless something changed the class you should be targetting for that page is .com_content.view-<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14501">issues.joomla.org/tracker/joomla-cms/14501</a>.</sub>

File: administrator/templates/isis/css/template-rtl.css [hunk 1]
```diff
@@ -8533,7 +8533,7 @@ td.nowrap.has-context {
 	border-right: 1px solid #b3b3b3;
 	height: auto;
 	line-height: inherit;
-	margin: 0 6px 0 -10px;
+	margin: 0 6px 0 0px;
```
Comment (brianteeman) [inline administrator/templates/isis/css/template-rtl.css]: shouldnt that be 0 6px 0 0
?

File: administrator/templates/isis/css/template-rtl.css [hunk 2]
```diff
@@ -8994,6 +8994,61 @@ input:focus,
 		column-count: auto;
 	}
 }
+@media (min-width: 1200px) {
+	.com_content.view-articles .subhead .container-fluid {
+		padding-right: 5px;
+		padding-left: 5px;
+	}
+	.com_content.view-articles .btn-toolbar .btn-wrapper {
+		margin: 0 0 8px 0px;
```
Comment (brianteeman) [inline administrator/templates/isis/css/template-rtl.css]: shouldnt that be 0 0 8px 0;

File: administrator/templates/isis/css/template.css [hunk 3]
```diff
@@ -8533,7 +8533,7 @@ td.nowrap.has-context {
 	border-right: 1px solid #b3b3b3;
 	height: auto;
 	line-height: inherit;
-	margin: 0 6px 0 -10px;
+	margin: 0 6px 0 0px;
```
Comment (brianteeman) [inline administrator/templates/isis/css/template.css]: shouldnt that be 0 6px 0 0;

File: administrator/templates/isis/css/template.css [hunk 4]
```diff
@@ -8994,3 +8994,58 @@ input:focus,
 		column-count: auto;
 	}
 }
+@media (min-width: 1200px) {
+	.com_content.view-articles .subhead .container-fluid {
+		padding-right: 5px;
+		padding-left: 5px;
+	}
+	.com_content.view-articles .btn-toolbar .btn-wrapper {
+		margin: 0 0 8px 0px;
```
Comment (brianteeman) [inline administrator/templates/isis/css/template.css]: shouldnt that be 0 0 8px 0;

File: administrator/templates/isis/less/blocks/_toolbar.less [hunk 5]
```diff
@@ -89,7 +89,7 @@
 	    border-right: 1px solid @btnBorder;
 	    height: auto;
 	    line-height: inherit;
-	    margin: 0 6px 0 -10px;
+	    margin: 0 6px 0 0px;
```
Comment (brianteeman) [inline administrator/templates/isis/less/blocks/_toolbar.less]: again 0px should be 0

File: administrator/templates/isis/less/pages/_com_content.less [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+	
+	/* Toolbar Container Responsive Adjustments */
+	@media (min-width: 1200px) {
+		.subhead .container-fluid {
+			padding-right:5px;
+			padding-left:5px;
+		}
+		 
+		.btn-toolbar .btn-wrapper {
+			margin: 0 0 8px 0px;
```
Comment (brianteeman) [inline administrator/templates/isis/less/pages/_com_content.less]: 0px should be 0

**Comment by cpfeifer** [comment]
Thank you for the input, I’ve made adjustments.

This does not need to be as complicated as I’ve made it here, we could simply make the buttons a touch smaller. It’s a styling issue which already exists in 3.6 that no one has been overly concerned with until now. That is one reason we don’t have a lot feedback on it, it’s either not being noticed or not many people are bothered by it. 

I am not assuming the buttons will be the same size in every language. I know that is a factor, but I only speak American. This was not taken into consideration with this redesign, nor was it was the design tested at smaller screen sizes. I was not included in this process at any time.

We certainly don’t need remove any buttons. This method is based on a theory of how important these buttons are to our users, they are assumptions made on the small amount feedback we have on this topic, we are planning to test this theory in more detail. We just got our official subdomain yesterday, as soon as it’s set up and configured we’ll be launching more studies. The toolbar will be among the first of them.

First to go is the edit button, which doesn’t seem to serve any purpose. It doesn’t have batch functionality and it adds an additional click the article edit process. Ciaran's solution to this is awesome and I think we should use it, either now or in J4.

From there, the publish / unpublish batch functionality seems to have limited real world applications, that is a theory, then archive and trash next. All 4 of these functions are available on a per item basis within the list. The only functionality change is the absence of one click batch functionality for these items.

The sort and item number display functions are already being removed in this way under desktop size. In my opinion, those controls are far more useful on smaller screens than batch buttons. That’s another theory and an entirely different topic, but there is already a precedent for removing controls at smaller screen sizes.

Bottom line, not much attention has been given to mobile styling in the 3.x backend. It essentially had bootstrap slapped on it with a mobile size menu and some other minor adjustments. That is not enough for an interface with this many controls. It’s been like this for years and no one has said much about it. 

Any changes made now will have a minimal impact on this release. J4 is the time to address this, we’ve got bigger fish to fry on J3.7.

**Comment by brianteeman** [comment]
You (UX team) were the one who said it was a high priority. I guess now you
are saying it's not and your not interested in 3.7 even though it has at
least 30 months of life left. Now I know why nothing ever happens with UX
in Joomla despite the huge amount of cash that has been spent on the UX team

**Comment by dgt41** [comment]
Other approaches:
https://github.com/joomla/joomla-cms/pull/13250
https://github.com/joomla/joomla-cms/issues/13230

File: administrator/templates/isis/less/pages/_com_content.less [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+		}
+	}
+	
+	/* Hide Edit, Featured & Unfeatured Under 980px */
+	@media (min-width: 481px) and (max-width: 979px) {
+		#toolbar-edit, #toolbar-featured, #toolbar-unfeatured, {display:none;}
+	}
+	
+	/* Hide Archive & Trash Under 767px */
+	@media (min-width: 481px) and (max-width: 768px) {
```
Comment (C-Lodder) [inline administrator/templates/isis/less/pages/_com_content.less]: max width for mobiles should be `767px`, not `768px`

File: administrator/templates/isis/css/template-rtl.css [hunk 8]
```diff
@@ -8994,6 +8994,61 @@ input:focus,
...
+	}
+}
+@media (min-width: 481px) and (max-width: 979px) {
+	.com_content #toolbar-edit,
+	.com_content #toolbar-featured,
+	.com_content #toolbar-unfeatured {
+		display: none;
+	}
+}
+@media (min-width: 481px) and (max-width: 768px) {
```
Comment (C-Lodder) [inline administrator/templates/isis/css/template-rtl.css]: max width for mobiles should be `767px`, not `768px`

File: administrator/templates/isis/less/pages/_com_content.less (hunk 7 above)
Comment (mbabker) [inline administrator/templates/isis/less/pages/_com_content.less]: Aren't there variables that can be used for these?

Comment (C-Lodder) [inline administrator/templates/isis/less/pages/_com_content.less]: Joomla + LESS? Variables? Good joke

Comment (mbabker) [inline administrator/templates/isis/less/pages/_com_content.less]: I woulda thought there was something in the Bootstrap 2 library with them.  If not, I've just been spoiled by UI frameworks with updates in 2016 then.

**Comment by cpfeifer** [comment]
We just received an official resource to do official research yesterday after a year of me asking for one. We were not included in this release process on any official level, and not much of the input we've provided has ever been taken into consideration in the development process. 

Asking for UX input at the Alpha stage is too late. A public release is the end of a user centered process, not the beginning. Joomla uses a development driven process and we haven't been given a fair chance do to anything as of yet. That's not something I can change alone, and we could do more if people wanted worked with us instead of throwing us under the bus every chance they get.

UI is a top UX priority for J4. We are very limited in what we can do for any version of 3.x, we can't just starting changing everything, nor should we.

This is a minor styling issue, let's not make into something it's not. We should fix the issue on move on. If there are constructive comments I'm happy to make adjustments. If there is a better solution than mine, let's use that one.

File: administrator/templates/isis/less/pages/_com_content.less (hunk 7 above)
Comment (C-Lodder) [inline administrator/templates/isis/less/pages/_com_content.less]: There are variables, not that many, but they do exists. I was talking more about Joomla's LESS being crap.

**Comment by C-Lodder** [comment]
@cpfeifer - https://github.com/joomla/joomla-cms/pull/14501#pullrequestreview-27398041

**Comment by dgt41** [comment]
Sorry @cpfeifer but **YOU ARE REMOVING functionality** here!!!
Either provide the proof to back it up or close the PR!

**Comment by brianteeman** [comment]
> Asking for UX input at the Alpha stage is too late. 

No it was asked for a very long time ago - for this specific issue on 14 November - a time when you yourself said 

> The UX team will review this issue asap, among many others. I've been tied up with JWC for the past month. Now it's over and I'll be home tomorrow night. We worked out a strategy over the weekend to have more streamlined involvement and input between our team and the development process across all 3.x & J4 issues. Just FYI

You cant pull the time/date card again

**Comment by cpfeifer** [comment]
As I said, we don't need to remove the buttons. The importance of these buttons is a theory we are planning to test. I can easily remove the last few lines, the buttons stay there, and the original issue is addressed.

I've said since day one this issue this could be addressed with simple CSS. I have looked into this extensively over the past few months and the mobile styling issues go far beyond the toolbar. Any viable long term solution includes an overhaul of the entire UI. The toolbar is critical piece of the interaction, but changing this one piece without addressing the other issues does not make any sense. It makes far more sense to do this in J4 than it does to try fix this mess now. 

One button being slightly more out place than it was before on one interface screen, at one screen size, does not equal "redo the entire toolbar immediately". It means adjust the design.

**Comment by brianteeman** [comment]
Why don't you just rename your team j4only and then the rest of joomla
contributors won't waste months waiting for a response to something you
clearly have no interest in.
